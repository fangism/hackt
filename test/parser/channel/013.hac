// "013.hac" should fail, currently forbid arrays in channel types

defchan mychan <: chan(int<16>[2]) (int<32> i);

