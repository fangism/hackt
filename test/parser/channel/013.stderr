At: "013.hac":3:
parse error: syntax error
parser stacks:
state	value
#STATE#	(null) 
#STATE#	(null) 
#STATE#	(null) 
#STATE#	keyword: defchan [3:1..7]
#STATE#	identifier: mychan [3:9..14]
#STATE#	<: [3:16]
#STATE#	keyword: chan [3:19..22]
#STATE#	( [3:23]
#STATE#	datatype: int [3:24..26]
#STATE#	list<(expr)>: (array-concatenation) ... [3:27..30]
#STATE#	[ [3:31]
in state #STATE#, possible rules are:
	strict_relaxed_template_arguments : complex_expr_optional_list_in_angles . optional_template_arguments_in_angles  (#RULE#)
	optional_template_arguments_in_angles : .  (#RULE#)
acceptable tokens are: 
	'!' (reduce)
	'(' (reduce)
	')' (reduce)
	',' (reduce)
	'<' (shift)
	'?' (reduce)
	ID (reduce)
